daemon/daemon_unix.go

```diff
@@ -433,6 +435,22 @@ func checkSystem() error {
 	return checkKernel()
 }
 
+// configureMaxThreads sets the Go runtime max threads threshold
+// which is 90% of the kernel setting from /proc/sys/kernel/threads-max
+func configureMaxThreads(config *Config) error {
+	mt, err := ioutil.ReadFile("/proc/sys/kernel/threads-max")
+	if err != nil {
+		return err
+	}
+	mtint, err := strconv.Atoi(strings.TrimSpace(string(mt)))
+	if err != nil {
+		return err
+	}
+	maxThreads := (mtint / 100) * 90
+	debug.SetMaxThreads(maxThreads)
```

Comment:
Could we get a log statement in here? I feel like it might help debugging if the threads happen to be set to some strange value to know what value it was set to.